### PR TITLE
use local id in the space range for tag edge index

### PR DESCRIPTION
**Pull request by panda-sheep**
As title
1. Tag, edge, tag index and edge index  use space range id.
2. Compatible with the existing old schema creating  previously.

### CODE REVIEW

File: src/meta/processors/BaseProcessor.inl [hunk 1]
```diff
@@ -188,6 +188,43 @@ ErrorOr<nebula::cpp2::ErrorCode, int32_t> BaseProcessor<RESP>::autoIncrementId()
...
+    folly::SharedMutex::WriteHolder holder(LockUtils::localIdLock());
+    auto localIdkey = MetaServiceUtils::localIdKey(spaceId);
+    int32_t id;
+    std::string val;
+    auto ret = kvstore_->get(kDefaultSpaceId, kDefaultPartId, localIdkey, &val);
+    if (ret != nebula::cpp2::ErrorCode::SUCCEEDED) {
+        if (ret != nebula::cpp2::ErrorCode::E_KEY_NOT_FOUND) {
+            return ret;
+        }
+        id = 1;
```
Comment (critical27) [inline src/meta/processors/BaseProcessor.inl:203]: You can't set to 1. This is a bit more complicated when there is old data, you need to think about it carefully. Better add a ut to mock this case as well.

Comment (panda-sheep) [reply src/meta/processors/BaseProcessor.inl:203]: When upgrading, when the current space has an old schema, 
you only need to find the maximum value of the tagId, edgetype, and indexid under the current space,
and the id is the maximum value +1
Then insert the key value into the meta.

This approach has changed very little.

Comment (panda-sheep) [reply src/meta/processors/BaseProcessor.inl:203]: But this needs to be done in the meta upgrade tool.

Comment (critical27) [reply src/meta/processors/BaseProcessor.inl:203]: `maximum value of old autoIncrementId +1` sounds ok. Meta is upgraded in-place, no extra tools. If you set to 1, the old data will be messed up.

Comment (panda-sheep) [reply src/meta/processors/BaseProcessor.inl:203]: Yes, I mean in-place updated meta.
Just deal with it here.

**Comment by bright-starry-sky** [comment]
I can't find where `autoIncrementIdInSpace` is used for `create tag | edge` .

**Comment by bright-starry-sky** [comment]
> I can't find where `autoIncrementIdInSpace` is used for `create tag | edge` .

This PR only deals with indexes, right?

**Comment by panda-sheep** [comment]
> > I can't find where `autoIncrementIdInSpace` is used for `create tag | edge` .
> 
> This PR only deals with indexes, right?

tag, edge, tag index. edge index are all processed
I will immediately add another test with an existing schema

**Comment by bright-starry-sky** [comment]
> > > I can't find where `autoIncrementIdInSpace` is used for `create tag | edge` .
> > 
> > 
> > This PR only deals with indexes, right?
> 
> tag, edge, tag index. edge index are all processed
> I will immediately add another test with an existing schema

yes,  I saw what processed. good job.

File: src/meta/processors/BaseProcessor.inl (hunk 1 above)
Comment (panda-sheep) [reply src/meta/processors/BaseProcessor.inl]: In order to be compatible with the existing old schema and implementation convenience, when the local_id key of the current space does not exist, it is directly set to autoIncrementId + 1.  thx

**Comment by panda-sheep** [comment]
Ready, please continue.

**Review by bright-starry-sky — DISMISSED** [review]
LGTM

**Comment by critical27** [comment]
LGTM, wait the version has been released.

**Pull request closed without merging**